omnibus/config/software/openssl-fips-provider.rb

```diff
@@ -0,0 +1,48 @@
+# Unless explicitly stated otherwise all files in this repository are licensed
+# under the Apache License Version 2.0.
+# This product includes software developed at Datadog (https://www.datadoghq.com).
+# Copyright 2016-present Datadog, Inc.
+
+# Embedded OpenSSL to meet FIPS requirements. It comes in two parts:
+# 1. The FIPS module itself (this software definition). It must use a FIPS-validated version 
+#    and follow the build steps outlined in the OpenSSL FIPS Security Policy.
+# 2. The OpenSSL library, which can be any 3.0.x version. This library will use the FIPS provider. 
+
+name "openssl-fips-provider"
+default_version "0.0.1"
+
+OPENSSL_FIPS_MODULE_VERSION="3.0.9"
+OPENSSL_FIPS_MODULE_FILENAME="openssl-#{OPENSSL_FIPS_MODULE_VERSION}.tar.gz"
+OPENSSL_FIPS_MODULE_SHA256_SUM="eb1ab04781474360f77c318ab89d8c5a03abc38e63d65a603cabbf1b00a1dc90"
+
+source url: "https://www.openssl.org/source/#{OPENSSL_FIPS_MODULE_FILENAME}",
+        sha256: "#{OPENSSL_FIPS_MODULE_SHA256_SUM}",
+        extract: :seven_zip,
+        target_filename: "#{OPENSSL_FIPS_MODULE_FILENAME}"
+
+relative_path "openssl-#{OPENSSL_FIPS_MODULE_VERSION}"
+
+build do
+    # Exact build steps from security policy:
+    # https://csrc.nist.gov/CSRC/media/projects/cryptographic-module-validation-program/documents/security-policies/140sp4282.pdf
+    #
+    # ---------------- DO NOT MODIFY LINES BELOW HERE ----------------
+    command "./Configure enable-fips"
+
+    command "make"
+    command "make install"
+    # ---------------- DO NOT MODIFY LINES ABOVE HERE ----------------
+
+    mkdir "#{install_dir}/embedded/ssl"
+    mkdir "#{install_dir}/embedded/lib/ossl-modules"
+    copy "/usr/local/lib*/ossl-modules/fips.so", "#{install_dir}/embedded/lib/ossl-modules/fips.so"
```

Comment:
Would it make sense to follow the instructions on https://github.com/openssl/openssl/blob/master/README-FIPS.md#copy-the-fips-provider-artifacts-fipsso--fipsmodulecnf-to-known-locations and use `make install_fips` after having configured the environment as for the rest of the software we build? Instead of doing essentially the same thing manually.

Reply:
I don't think so. we have the separate `fipsinstall.sh` to run Step 2 of [the validated 140-2 openssl fips provider](https://csrc.nist.gov/CSRC/media/projects/cryptographic-module-validation-program/documents/security-policies/140sp4282.pdf) installation instructions. (see appendix A, pg 34)

make install_fips [copies](https://github.com/openssl/openssl/blob/47a80fd2034cd4314d3b4958539dcd3106087109/Configurations/unix-Makefile.tmpl#L667-L679) but does not run the module's self tests or generate the config file output around the self-tests and checksum AFAICT